src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/BitbucketWebhookHandler.java

```diff
@@ -108,15 +110,14 @@ private void deleteWebhooks(List<BitbucketWebhook> webhooks) {
      * @return the correct webhook event
      */
     private Collection<BitbucketWebhookEvent> getEvents(WebhookRegisterRequest request) {
-        // TODO: logging for when some events are not supported
         Collection<BitbucketWebhookEvent> supportedEvents = new HashSet<>();
         if (request.isMirror()) {
             try {
                 BitbucketWebhookSupportedEvents events = serverCapabilities.getWebhookSupportedEvents();
-                Set<String> hooks = events.getApplicationWebHooks();
-                if (hooks.contains(MIRROR_SYNCHRONIZED_EVENT.getEventId())) {
+                Set<String> hooks = events.getApplicationWebHooks(); //set intersection, cast list to set set.unison
+                if (!hooks.stream().filter(MIRROR_SYNCHRONIZED_EVENT.getEventIds()::contains).collect(Collectors.toSet()).isEmpty()) {
```

Comment:
maybe we could do 
```
if (!Collections.disjoint(hooks, MIRROR_SYNCHRONIZED_EVENT.getEventIds())) {
```
instead? does that read better? I think it is clearer what you're trying to do? Maybe other reviewers can comment and see what they think is clearer?